### ISSUE TITLE
What are the information necessary to reconciliate the pseudonymization?

### ISSUE THREAD

**Issue by Chardonneaur** [piwik/piwik#19687]
When using the user id pseudonymization feature, it is written that some additional information may be necessary in order to re conciliate the user ID.
It would be interesting to have more information about it, are we referring to Matomo default information (user agent for example) and in this case to link to a specific page with more information, or are we referring to the client data information (internal database for example orders).
Thank you very much.

**Comment by sgiehl** [comment]
@Chardonneaur I guess @tsteur might be able to answer this best. But from my understanding the pseudonymized user id is calculated using the userid and a configured salt (which differs from instance to instance) in a one way hash method. So if you know the salt and someones user id, you are easily able to generate the pseudonymized user id and would be able to easily look up all their visits.

**Comment by heurteph-ei** [comment]
@Chardonneaur
I think also that you can identify easily somebody with pseudonymization: I use it to find myself.
I browse at a certain time. Then I go in the Matomo dashboard, and look for my visit. Get the User ID (pseudonymized). That's it, I am no more anonymous at all. (And even, I create some segments on myself ;-) )

**Comment by Chardonneaur** [comment]
Ok clearer, thank you, to me that's clearly a page to write about within the FAQ to explain exactly this.

**Comment by tsteur** [comment]
What @sgiehl mentioned is correct 👍  There's a unique hash being generated that is used for hashing userIDs. And what @heurteph-ei mentions is correct too. If you can cross link a visit with other information (such as when they contacted you, or ordered from you, or ...), then you can also identify an individual through the hashed userId.

**Comment by Chardonneaur** [comment]
I agree, i just would like to clarify that I opened this ticket in order to say "hey, that will be great to describe this as an FAQ entry".

**Comment by MatomoForumNotifications** [comment]
This issue has been mentioned on **Matomo forums**. There might be relevant details there:

https://forum.matomo.org/t/is-consent-exemption-possible-with-an-ecommerce-website/46778/4
